lib/arel-helpers/join_association.rb

```diff
@@ -108,7 +108,7 @@ def join_association_5_0(table, association, join_type)
         constraints = join_dependency.join_constraints([], join_type)
 
         binds = constraints.flat_map do |info|
-          prepared_binds = table.connection.prepare_binds_for_database(info.binds)
+          prepared_binds = info.binds.map(&:value_for_database)
```

Comment:
@camertron This change comes from this PR https://github.com/rails/rails/pull/25937/files